internal/datastore/postgres/revisions.go

```diff
@@ -56,7 +56,14 @@ const (
 	//   %[5] Name of the snapshot column
 	queryValidTransaction = `
 	WITH minvalid AS (
-		SELECT %[1]s, %[5]s FROM %[2]s WHERE %[3]s >= NOW() - INTERVAL '%[4]f seconds' ORDER BY %[3]s ASC LIMIT 1
+		SELECT %[1]s, %[5]s
+        FROM %[2]s
+        WHERE 
+            %[3]s >= NOW() - INTERVAL '%[4]f seconds'
+          OR
+             %[3]s = (SELECT MAX(%[3]s) FROM %[2]s)
```

Comment:
not blocking, but if you end up touching this PR again the spacing here is slightly off